src/mame/nintendo/vt1682.cpp

```diff
@@ -5353,8 +5444,19 @@ uint8_t vt_vt1682_state::soundcpu_irq_vector_hack_r(offs_t offset)
 
 uint8_t vt_vt1682_state::maincpu_irq_vector_hack_r(offs_t offset)
 {
-	// redirect to Timer IRQ!
-	return rom_8000_to_ffff_r((0xfff8 - 0x8000)+offset);
+	if (!machine().side_effects_disabled())
+	{
+		if (offset == 0)
+			m_current_main_vector = m_new_main_vector;
+
+		uint8_t ret = rom_8000_to_ffff_r((m_current_main_vector - 0x8000) + offset);
+		// redirect to required IRQ!
+		return ret;
+	}
+	else
+	{
+		return rom_8000_to_ffff_r((0xfffe - 0x8000) + offset);
+	}
 }
```

Comment:
This isn’t going to show the value that will be read in the debugger, which is confusing.  You’re going to need to do something like:
```c++
	auto vector = m_current_main_vector;
	if (offset == 0)
	{
		vector = m_new_main_vector;
		if (!machine().side_effects_disabled())
			m_current_main_vector = vector;
	}

	// redirect to required IRQ!
	return rom_8000_to_ffff_r((vector - 0x8000) + offset);
```

Reply:
well I intentionally coded it because I wanted to see the value in actual ROM space there, that was a choice

Reply:
I'd prefer to leave this as it is, the vectors can be viewed from the drop down menu in the memory viewer anyway, and I think showing a modified version of ROM is more confusing.

Reply:
Viewing the address space is supposed to show what would be read from that address.  It’s confusing for developers if viewing the address space or reading from the address space in a Lua script or debugger script doesn’t  If you want to view ROM, you can view the ROM region in the debugger.

Reply:
I still think you're overstepping the mark here, I developed this in the way that is least confusing to me, the developer of the driver.  I find what you're suggesting more confusing, and do not wish to change it to be like that.

Again you're coming off as incredibly patronizing here.

Reply:
Anyway, I've made my point above, I won't be changing this.  It is more beneficial the way I have it.  Does that mean I should just close this whole PR and abandon work as it is never going to be merged as per seems the way around here whenever anybody disagrees with you?

Reply:
Suppressing side effects is supposed to do what it says, i.e. suppress side effects.  If it doesn’t do that, it breaks expectations and confuses people.  When you’re debugging, viewing the address space should show what the CPU will see if it reads any given location.  Side effects have to be suppressed to make that non-intrusive.

Returning a different value for the debugger and scripts breaks that expectation.  At best it will leave people asking, “Why isn’t the CPU seeing the value I’m seeing?”  But more likely it will just leave them confused that the CPU is doing something strange with the value it reads because they won’t consider that the driver would be coded in such a confusing way.

If someone attempting to debug code running on an emulated system in MAME has to wade through MAME’s source to work out why the value read by the CPU isn’t what they see in the address space view in the debugger, that’s broken.

> It is more beneficial the way I have it.

Breaking expectations is not beneficial.  It makes MAME confusing and unapproachable.

> Does that mean I should just close this whole PR and abandon work as it is never going to be merged as per seems the way around here whenever anybody disagrees with you?

If you want to throw a tantrum, fine.

Reply:
Once the logic is moved to the CPU core the debugger view will be identical to what you see now.

This CPU pulls vectors from difference places, even on the sound CPU side there is no vector mirror at fffx from what people have told me, it simply pulls from 3ffx or 1ffx depending on if internal ROM is enabled or not.

With the IRQ controller in the CPU core, there will be no overlay, it will pull the vectors from the address associated with that interrupt level.

The move to the CPU core is (was) scheduled for a later PR.  Why you insist on this temporary (incorrect) view an interfering with driver development I do not know.

Just don't complain about having too much work to do when you won't allow devs to actually develop without interfering.

Feedback is meant to be a 2 way process, mentioning things that the author might have missed.  In this case it is not something I missed, it is an intentional design.  If you can't respect your developers you're going to very quickly be left with none.

Reply:
If you want this to go in, change it to do what Vas said.  You're perfectly free to make your local copy do the weird stuff again afterwards, but what lands in master has to make sense, and the debugger not showing what the CPU actually sees qualifies as not making sense.

Reply:
The real CPU sees what the debugger shows now.  It doesn't see the overlay.

I will not be changing this.

Reply:
If somebody merges this, and the work gets done on the CPU core to just fetch from the proper address it's going to look exactly like it looks now, does that get blocked too?

If you know this uses the VT SoC then you know where to look for the correct vectors, you don't want to see the overlay in the debugger.

Reply:
I can't stop somebody else from making the change post-merge, which honestly is the most likely thing to happen here, because there's absolutely no reason I should be bending to demands to make the code I have less correct.

I can still not appreciate that level of interference and micro-management and act accordingly.

Reply:
As a compromise I've added a comment explaining why that logic is correct, and why it is as it is.